app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/SingleCardTest.kt

```diff
@@ -0,0 +1,268 @@
+    private val invalidJson = "This is not a valid json string"
+
+    //ViewMatchers of the fragment
+    private val textCardName = onView(withId(R.id.singleCard_text_cardName))
+    private val textCardSet = onView(withId(R.id.singleCard_text_cardSet))
+    private val textCardNumber = onView(withId(R.id.singleCard_text_cardNumber))
+    private val textCardText = onView(withId(R.id.singleCard_text_cardText))
+    private val textCardRarity = onView(withId(R.id.singleCard_text_cardRarity))
+    private val textCardType = onView(withId(R.id.singleCard_text_cardTypeSubtypesStats))
+    private val imageCard = onView(withId(R.id.singleCard_image))
+    private val textCardArtist = onView(withId(R.id.singleCard_text_cardArtist))
+    private val textCardManaCost = onView(withId(R.id.singleCard_text_cardManaCost))
+
+    @Before
+    fun setup() {
+        IdlingRegistry.getInstance().register(CustomGlide.countingIdlingResource)
+    }
+
+    @After
+    fun tearDown() {
+        IdlingRegistry.getInstance().unregister(CustomGlide.countingIdlingResource)
+        scenario.close()
+    }
+
+    /**
+     * This test is used to check if the fragment is able to manage
+     * the case where no arguments are passed to the fragment.
+     */
+    @Test
+    fun testNoJsonInArgumentsIsInvalid() {
+        scenario = launchFragmentInContainer()
+
+        textCardName.check(matches(withText(R.string.error_load_card)))
+        textCardSet.check(matches(withText("")))
+        textCardNumber.check(matches(withText("")))
+        textCardText.check(matches(withText("")))
+        textCardRarity.check(matches(withText("")))
+        textCardType.check(matches(withText("")))
+        textCardArtist.check(matches(withText("")))
+        textCardManaCost.check(matches(withText("")))
+        imageCard.check(matches(not(isDisplayed())))
+    }
+
+    /**
+     * This test is used to check if the fragment is able to manage
+     * the case where invalid arguments are passed to the fragment.
+     * For example : a json string that is not a MagicCard.
+     */
+    @Test
+    fun testInvalidJsonInArgumentsIsInvalid() {
+        val bundleArgs = Bundle().apply { putString("card", invalidJson) }
+        scenario = launchFragmentInContainer(fragmentArgs = bundleArgs)
+
+        textCardName.check(matches(withText(R.string.error_load_card)))
+        textCardSet.check(matches(withText("")))
+        textCardNumber.check(matches(withText("")))
+        textCardText.check(matches(withText("")))
+        textCardRarity.check(matches(withText("")))
+        textCardType.check(matches(withText("")))
+        textCardArtist.check(matches(withText("")))
+        textCardManaCost.check(matches(withText("")))
+        imageCard.check(matches(not(isDisplayed())))
+    }
+
+    /**
+     * This test is used to check if the fragment is able to display a valid card.
+     */
+    @Test
+    fun testValidJsonInArgumentsIsValid() {
+        val bitmap = Glide.with(context)
```

Comment:
maybe worth to mock this ? download once the image, put it in ressources and then load it for the test from the local assets for testing

Reply:
That would indeed be a possibility. I may be less of a fan of the network anyway to download the first image, and that this test may be automatically modified depending on the imageUrl. If you tell me there is a noticeable difference, I can make the change.

Reply:
not sure indeed of the noticeable difference, i like the way to use the actual url to keep the image updated, but it means we are actually testing the network as well. If we are testing the display of the card, we can subsitute with our own resources instead, using a local bitmap + tricking the card to not use the actual url to download the image but to display the given bitmap. This require more work and can be done later on because you need the mockWebserver integration.
Leave it like this for now, just put a todo for future modification.